airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/normalization/NormalizationLogParser.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.normalization;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.annotations.VisibleForTesting;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.integrations.destination.normalization.SentryExceptionHelper.ErrorMapKeys;
+import io.airbyte.protocol.models.AirbyteErrorTraceMessage;
+import io.airbyte.protocol.models.AirbyteErrorTraceMessage.FailureType;
+import io.airbyte.protocol.models.AirbyteLogMessage;
+import io.airbyte.protocol.models.AirbyteLogMessage.Level;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.AirbyteMessage.Type;
+import io.airbyte.protocol.models.AirbyteTraceMessage;
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.apache.logging.log4j.util.Strings;
+
+/**
+ * A simple wrapper for base-normalization logs. Reads messages off of stdin and sticks them into
+ * appropriate AirbyteMessages (log or trace), then dumps those messages to stdout
+ * <p>
+ * does mostly the same thing as
+ * {@link io.airbyte.workers.normalization.NormalizationAirbyteStreamFactory}. That class is not
+ * actively developed, and will be deleted after all destinations run normalization in-connector.
+ * <p>
+ * Aggregates all error logs and emits them as a single trace message at the end. If the underlying
+ * process emits any trace messages, they are passed through immediately.
+ */
+public class NormalizationLogParser {
+
+  private final List<String> dbtErrors = new ArrayList<>();
+
+  public Stream<AirbyteMessage> create(final BufferedReader bufferedReader) {
+    return bufferedReader.lines().flatMap(this::toMessages);
+  }
+
+  public List<String> getDbtErrors() {
+    return dbtErrors;
+  }
+
+  @VisibleForTesting
+  Stream<AirbyteMessage> toMessages(final String line) {
+    if (Strings.isEmpty(line)) {
+      return Stream.of(logMessage(Level.INFO, ""));
+    }
+    final Optional<JsonNode> json = Jsons.tryDeserialize(line);
+    if (json.isPresent()) {
+      return jsonToMessage(json.get());
+    } else {
+      return nonJsonLineToMessage(line);
+    }
+  }
+
+  /**
+   * Wrap the line in an AirbyteLogMessage, and do very naive dbt error log detection.
+   * <p>
+   * This is needed for dbt < 1.0.0, which don't support json-format logs.
+   */
+  private Stream<AirbyteMessage> nonJsonLineToMessage(final String line) {
+    // Super hacky thing to try and detect error lines
+    if (line.contains("[error]")) {
```

Comment:
will this ever be ERROR? Do we need case insensitivity?

Reply:
I sure hope not... this logic is mostly copied from https://github.com/airbytehq/airbyte/blob/a6d6b063037b62574143e1d09af6b342cf8fc240/airbyte-connector-test-harnesses/acceptance-test-harness/src/main/java/io/airbyte/workers/normalization/NormalizationAirbyteStreamFactory.java#L77

de facto - our current pre-1.0 dbt versions all use the exact string `[error]`, and >=1.0 dbt supports json-formatted logs. So I think we'll get away with it.